### PR TITLE
Fix undefined `meta` for `str.split(expand=False)`

### PR DESCRIPTION
**Pull request by bnaul**
Looks like this got broken in #4744.

### CODE REVIEW

File: dask/dataframe/accessor.py [hunk 1]
```diff
@@ -147,6 +147,8 @@ def split(self, pat=None, n=-1, expand=False):
             else:
                 meta = type(self._series._meta)([" ".join(["a"] * 2 * n)])
                 meta = meta.str.split(n=n, expand=expand, pat=pat)
+        else:
+            meta = list
```
Comment (TomAugspurger) [inline dask/dataframe/accessor.py]: Ah, actually, can you try `meta = (self._series.name, list)` here? Will that work?

Right now we're dropping Series.name, if it has one.

File: dask/dataframe/tests/test_dataframe.py [hunk 2]
```diff
@@ -4062,6 +4062,14 @@ def test_map_partitions_delays_lists():
     assert any(str(L) == str(v) for v in out.__dask_graph__().values())
 
 
+def test_str_noexpand():
+    s = pd.Series(["a b c d", "aa bb cc dd", "aaa bbb ccc dddd"])
```
Comment (TomAugspurger) [inline dask/dataframe/tests/test_dataframe.py]: Can you add `name='foo'` here too? Ensure that it stays.

Comment (bnaul) [reply dask/dataframe/tests/test_dataframe.py]: Done

File: dask/dataframe/accessor.py (hunk 1 above)
Comment (bnaul) [reply dask/dataframe/accessor.py]: `list` didn't work but `object` did :+1:

**Pull request merged**